### PR TITLE
Restructure Repository

### LINKED ISSUE #71: Restructure repository to be more logical and useable
We should restructure to:

- separate CAM code from the core YOG parameterisation - see #67
- Avoid duplication of code or dual-maintained copies of files across the repository
  - Use references between directories
  - Move to CMake instead of Make to handle dependencies
- Move tests to their own location
  - Provide the option for users to specify what tests are built

This will help towards, but not fully resolve #69

### PR DESCRIPTION
**Pull request by jatkinson1000**
This should close #71 and #67 

- [x] separate CAM code from the core YOG parameterisation - see #67
- [x] Avoid duplication of code or dual-maintained copies of files across the repository
  - Use references between directories
  - Move to CMake instead of Make to handle dependencies
- [x] Move tests to their own location
  - [x] Provide the option for users to specify what tests are built

### CODE REVIEW

**Comment by jatkinson1000** [comment]
Note that I have also removed the SAM interface file that was under development (see https://github.com/m2lines/convection-parameterization-in-CAM/pull/72/commits/2b3adf3a668ed67ded0773bc27a04d98230235e9).

This is because it is now deprecated and is no longer being pursued.
If returning it can be recovered from git history.

**Comment by jatkinson1000** [comment]
One question I have is should we perhaps rename "NN_module" to "YOG_convection"?

This would be much clearer.
If I recall NN_Module was only ever a placeholder until someone told Jim and I what the "neural net we extracted to a module" shoudl be called.

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    set(NETCDF_F90 "YES")
+    find_package(NetCDF REQUIRED)
+endif()
+pkg_search_module(NETCDF_C netcdf)
+if (NETCDF_C_FOUND)
+    list(APPEND NETCDF_LIBRARIES "${NETCDF_C_LDFLAGS}")
+    list(APPEND NETCDF_INCLUDES "${NETCDF_C_INCLUDE_DIRS}")
+endif()
+
+# set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../)
```
Comment (MarionBWeinzierl) [inline tests/CMakeLists.txt:40]: Can this commented-out line be removed?

File: tests/cmake/FindNetCDF.cmake [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# - Find NetCDF
```
Comment (MarionBWeinzierl) [inline tests/cmake/FindNetCDF.cmake:1]: Is this needed?

**Review by MarionBWeinzierl — APPROVED** [review]
We went through this together this morning and this makes sense. I have some very minor comments below, but otherwise it can be merged.

Regarding the question about the renaming, I agree that "YOG_convection" is a more meaningful name -- this change could be included in this PR

File: tests/cmake/FindNetCDF.cmake (hunk 2 above)
Comment (jatkinson1000) [reply tests/cmake/FindNetCDF.cmake:1]: If I remove it on my system it still works, but my recollection is that on larger more diverse systems (HPCs) it is required. I think leave in for now.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (jatkinson1000) [reply tests/CMakeLists.txt]: Yes. Done in https://github.com/m2lines/convection-parameterization-in-CAM/pull/72/commits/d8e8caf51bd3fe7032ccd690665f9970eb4c6ebd

**Comment by jatkinson1000** [comment]
Hi @MarionBWeinzierl Made those changes as suggested.
I'll go ahead and merge shortly.

**Pull request merged**